modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala

```diff
@@ -110,7 +109,7 @@ object BlockConsensusCell {
       _ <- cleanUpRoundData(ownProposal, ctx)
     } yield ()
 
-  private def broadcast[F[_]: Async, T <: Transaction: Encoder](
+  private def broadcast[F[_]: Async, T <: Transaction](
```

Comment:
Where is that Encoder for `T <: Transaction` take from right now?

Reply:
hmm, good question. I see that the `PeerBlockConsensusInput` requires implicit encoder/decoder

Reply:
but looks like the tests are passing, so consensus and transactions are working correctly 🤔

Reply:
For `DAGTransaction` it is derevo's `Encoder[DAGTransaction]`
<img width="193" alt="image" src="https://github.com/Constellation-Labs/tessellation/assets/9077743/2ea1f609-e3a2-4781-9131-50c48709ad52">

For `CurrencyTransaction` as well:
<img width="183" alt="image" src="https://github.com/Constellation-Labs/tessellation/assets/9077743/04eb7f0c-d4db-45cd-9adb-f758e545aabe">

Reply:
Ok the `BlockConsensusClient` has the `Encoder` for that given transaction type injected during creation.

Reply:
Yeah, the decoder is injected at the very top in the `Main` method. There are no intermediate instances.